crates/bevy_animation_rig/src/skinned_mesh.rs

```diff
@@ -0,0 +1,215 @@
+/// use bevy_ecs::{entity::Entity, system::{Commands, ResMut}};
+/// use bevy_math::Mat4;
+/// use bevy_pbr::prelude::PbrBundle;
+/// use bevy_render::pipeline::{RenderPipeline, RenderPipelines};
+///
+/// fn example_system(mut commands: Commands, mut skinned_mesh_inverse_bindposes_assets: ResMut<Assets<SkinnedMeshInverseBindposes>>) {
+///     // A skeleton with only 2 joints
+///     let skinned_mesh_inverse_bindposes = skinned_mesh_inverse_bindposes_assets.add(SkinnedMeshInverseBindposes(vec![
+///         Mat4::IDENTITY,
+///         Mat4::IDENTITY,
+///     ]));
+///
+///     // The inverse bindposes then can be shared between multiple skinned mesh instances
+///     for _ in 0..3 {
+///         commands.spawn_bundle(PbrBundle {
+///             render_pipelines: RenderPipelines::from_pipelines(
+///                 vec![RenderPipeline::new(SKINNED_MESH_PIPELINE_HANDLE.typed())]
+///             ),
+///             ..Default::default()
+///         }).insert(SkinnedMesh::new(
+///             skinned_mesh_inverse_bindposes.clone(),
+///             // Remember to assign joint entity here!
+///             vec![Entity::new(0); 2],
+///         ));
+///     }
+/// }
+/// ```
+#[derive(Debug, TypeUuid)]
+#[uuid = "b9f155a9-54ec-4026-988f-e0a03e99a76f"]
+pub struct SkinnedMeshInverseBindposes(pub Vec<Mat4>);
+
+pub fn skinned_mesh_setup(
+    mut pipelines: ResMut<Assets<PipelineDescriptor>>,
+    mut shaders: ResMut<Assets<Shader>>,
+    mut render_graph: ResMut<RenderGraph>,
+) {
+    let mut skinned_mesh_pipeline = pipelines
+        .get(render_graph::PBR_PIPELINE_HANDLE)
+        .unwrap()
+        .clone();
+    skinned_mesh_pipeline.name = Some("Skinned Mesh Pipeline".into());
+    skinned_mesh_pipeline.shader_stages.vertex = shaders.add(Shader::from_glsl(
+        ShaderStage::Vertex,
+        include_str!("skinned_mesh.vert"),
+    ));
+    pipelines.set_untracked(SKINNED_MESH_PIPELINE_HANDLE, skinned_mesh_pipeline);
+
+    render_graph.add_system_node(
+        "JointTransforms",
+        RenderResourcesNode::<SkinnedMesh>::new(false),
+    );
+    render_graph
+        .add_node_edge("JointTransforms", node::MAIN_PASS)
+        .unwrap();
+}
+
+pub fn skinned_mesh_update(
+    skinned_mesh_inverse_bindposes_assets: Res<Assets<SkinnedMeshInverseBindposes>>,
+    global_transform_query: Query<&GlobalTransform>,
+    mut skinned_mesh_query: Query<&mut SkinnedMesh>,
+) {
+    for mut skinned_mesh in skinned_mesh_query.iter_mut() {
+        let skinned_mesh_inverse_bindposes = skinned_mesh_inverse_bindposes_assets
+            .get(skinned_mesh.inverse_bindposes.clone())
+            .unwrap();
+
+        for i in 0..skinned_mesh.joint_entities.len() {
+            let global_transform = global_transform_query
+                .get(skinned_mesh.joint_entities[i])
+                .unwrap();
+            skinned_mesh.joint_transforms[i] =
+                global_transform.compute_matrix() * skinned_mesh_inverse_bindposes.0[i];
+        }
```

Comment:
Depending on if you decide to group the two vectors this could be simplified a bit. 

If not, you could always do a `zip` then `enumerate` of the the two separate vectors.

Reply:
I modified and moved the code to `update_joint_transforms()` in `impl SkinnedMesh`, you can review it there. I had to mutably borrow `SkinnedMesh` variable inside the scope of that function.